packages/mobile/src/pincode/PincodeSet.tsx

```diff
@@ -107,11 +134,25 @@ export class PincodeSet extends React.Component<Props, State> {
   onCompletePin2 = async (pin2: string) => {
     const { pin1 } = this.state
     if (this.isPin1Valid(pin1) && this.isPin2Valid(pin2)) {
-      setCachedPin(DEFAULT_CACHE_ACCOUNT, pin1)
       this.props.setPincode(PincodeType.CustomPin)
-      ValoraAnalytics.track(OnboardingEvents.pin_set)
+      if (this.isChangingPin()) {
+        const updated = await updatePin(this.props.account, this.state.oldPin, pin2)
+        if (updated) {
+          ValoraAnalytics.track(SettingsEvents.change_pin_new_pin_confirmed)
+          Logger.showMessage(i18n.t('accountScreen10:pinChanged'))
```

Comment:
It seems we should also `setCachedPin(DEFAULT_CACHE_ACCOUNT, pin1)` when updated, right?
otherwise the old pin would remain in cache.

Reply:
The `updatePin` function manages the cache, it clears it and saves the new pin there.